faiss/python/class_wrappers.py

```diff
@@ -226,7 +226,10 @@ def replacement_add(self, x):
 
         n, d = x.shape
         assert d == self.d
-        x = np.ascontiguousarray(x, dtype='float32')
+        if numeric_type == faiss.Float32:
```

Comment:
It is a bit clumsy that the Python interface is not able to directly accept `np.float16` arguments because there is no way to tell if this will raise an error.

Reply:
It should be a numpy array, see https://github.com/facebookresearch/faiss/blob/main/faiss/python/swigfaiss.swig#L1163

Support for torch arrays is via another mechanism.